Session-6/DeckOfCards/CardsDeck.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace DeckOfCards
+{
+    public abstract class CardsDeck<T>
+    {
+        public List<T> cardsDeck;
+        public CardsDeck()
+        {
+            cardsDeck = new List<T>();
+        }
+        public abstract void CreateDeck();
+
+        public void ShuffleDeck()
+        {
+            Random random = new Random();
+            for(int i = 0; i < cardsDeck.Count(); i++)
```

Comment:
Not a good shuffle algorithm. Random numbers can be repeated with the below logic.